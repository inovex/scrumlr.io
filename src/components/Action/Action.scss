@import '../../constants/style';

.action-wrapper {
  display: flex;
  justify-content: center;
}

.action {
  box-sizing: border-box;
  align-self: center;
  width: 100%;
  height: $default-padding * 2 + 32px;
  max-height: $default-padding * 2 + 32px;
  padding: $default-padding;
}

@media #{$desktop} {
  .action {
    padding: $default-padding $large-padding;
  }
}

.action-wrapper--theme-light {
  color: $color-dark-gray;
  background: $color-white;
}

.action-wrapper--theme-dark {
  color: $color-white;
  background: $color-dark-gray;
}

.action-wrapper--theme-mint {
  color: $color-white;
  background: $color-mint;
}