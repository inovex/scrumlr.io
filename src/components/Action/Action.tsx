import * as cx from 'classnames';
import * as React from 'react';

import './Action.scss';

export type ActionTheme = 'light' | 'dark' | 'mint';

export interface ActionProps {
  theme: ActionTheme;
  className?: string;
  children?: React.ReactNode;

  [key: string]: any;
}

const Action: React.SFC<ActionProps> = ({ theme, className, children }) => {
  const componentClassName = cx(
    'action-wrapper',
    `action-wrapper--theme-${theme}`,
    className
  );

  return (
    <div className={componentClassName}>
      <div className="action">{children}</div>
    </div>
  );
};

export default Action;
