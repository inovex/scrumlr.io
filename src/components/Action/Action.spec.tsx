import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { default as Action, ActionProps } from './Action';

describe('<Action />', () => {
  let wrapper: ShallowWrapper<ActionProps, {}>;

  it('should match snapshot', () => {
    wrapper = shallow(<Action theme="mint">Test</Action>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should extend list of classes when className is set', () => {
    const className = 'expected-test-class';
    wrapper = shallow(
      <Action theme="light" className={className}>
        Test
      </Action>
    );
    expect(wrapper.hasClass(className)).toBeTruthy();
  });
});
