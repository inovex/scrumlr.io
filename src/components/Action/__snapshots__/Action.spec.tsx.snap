// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> should match snapshot 1`] = `"<div class=\\"action-wrapper action-wrapper--theme-mint\\"><div class=\\"action\\">Test</div></div>"`;
